packages/react-ui/lib/__tests__/needsPolyfillPlaceholder-test.ts

```diff
@@ -0,0 +1,18 @@
+import { needsPolyfillPlaceholder } from '../needsPolyfillPlaceholder';
+import { isIE11 } from '../client';
+
+describe('needsPolyfillPlaceholder', () => {
+  if (isIE11) {
+    it('returns true if the browser is IE11', () => {
+      expect(needsPolyfillPlaceholder()).toEqual(true);
+    });
+  } else if ('placeholder' in document.createElement('input')) {
+    it('returns true if browser does not support placeholder attribute', () => {
+      expect(needsPolyfillPlaceholder()).toEqual(true);
+    });
+  } else {
+    it('returns false if browser supports placeholder attribute', () => {
+      expect(needsPolyfillPlaceholder()).toEqual(true);
+    });
+  }
```

Comment:
В текущем варианте всегда будет запускаться только один тест.
То есть мы не узнаем падает ли тест со вторым условием, если первое правдиво.
```suggestion
  it('returns true if the browser is IE11', () => {
    if (isIE11) {
      expect(needsPolyfillPlaceholder()).toEqual(true);
    }
  });

  it('returns true if browser does not support placeholder attribute', () => {
    if (!supportsPlaceholder) {
      expect(needsPolyfillPlaceholder()).toEqual(true);
    }
  });

  it('returns false if browser supports placeholder attribute', () => {
    if (isIE11 && !supportsPlaceholder) {
      expect(needsPolyfillPlaceholder()).toEqual(true);
    }
  });
```

Для изменений выше придётся ещё вытащить из функции и экспортировать наружу `supportsPlaceholder`.

```ts
// lib/needsPolyfillPlaceholder.ts

export const supportsPlaceholder = 'placeholder' in document.createElement('input');
```